modules/Bio/EnsEMBL/Hive/DBSQL/AccumulatorAdaptor.pm

```diff
@@ -47,6 +47,32 @@ sub default_table_name {
 }
 
 
+sub slicer {    # take a slice of the hashref (if only we could inline in Perl!)
+    my ($self, $hashref, $fields) = @_;
+
+    my $overflow_limit = $self->overflow_limit();
+
+    return [ map { eval { my $value = $hashref->{$_};
+                          my $ol = $overflow_limit->{$_};
+                          if (defined($ol) and defined($value) and  (length($value) > $ol)) {
+                              $self->db->get_AnalysisDataAdaptor()->store_if_needed($value);
+                          } else {
+                              $value;
+                          }
+                   }
+
+
+             } @$fields ];
+}
```

Comment:
Did you override `slicer` here because AccumulatorAdaptor is the only NakedTableAdaptor that has overflowing and you wanted to keep NakedTableAdaptor's `slicer` simple and straightforward ?

It just felt surprising at the first glance. Maybe you can add a comment on top of NakedTableAdaptor::slicer indicating that should we need to move the overflow capability up, we can find it in AccumulatorAdaptor ?

Reply:
Yes - I didn't want to impose the overhead on every inheriting Adaptor. However, the overhead is small, and for maintainability it's probably better to move it to the base class